### ISSUE TITLE
Deploy k8s-cluster without VMware compute cluster

### ISSUE THREAD

**Issue by semaph0r** [sguyennet/terraform-vsphere-kubespray#25]
Is it possible to use this pipelines on VMware without the need of a compute cluster and all the vMotion stuff? I don't need that.

I've looked into `vsphere-kubespray.tf` and it seems that the compute cluster variable is only used to navigate through the VCenter ressource graph more efficiently.

**Comment by sguyennet** [comment]
Hi,
Do you mean deploying Kubernetes on standalone ESXi (without a vCenter)?

**Comment by Harsun** [comment]
Hello,
The thing is, terraform is just an API client, pretty much modular.
So, in your case, if you wish you can create your own VMs without
Terraform. You can try to use ansible to create your own VMs and customize
them.
Or you can check other Terraform Modules, I didn't check but maybe someone
created Terraform modules for ESXi nodes directly.


*Karakas, Gokhan*

Freelance IT advisor, Security Investigator
*Mobile Turkey: +90 506 904 90 59*

**Comment by semaph0r** [comment]
> 
> 
> Hi,
> Do you mean deploying Kubernetes on standalone ESXi (without a vCenter)?

No, I am using vCenter but did not licenzed vMotion, so compute clusters are not available. Actually I do not plan to use compute clusters either. My questions is, if there are any side effects by refactor the scripts not to use compute clusters.

The refactoring would remove all compute cluster resources and it's references from the terraform script and should tie all subsequent resources directly together.

As of now, the only places where the compute cluster is use are:

```
resource "vsphere_resource_pool" "resource_pool" {
  name                    = "${var.vsphere_resource_pool}"
  parent_resource_pool_id = "${data.vsphere_compute_cluster.cluster.resource_pool_id}"
}
```
```
# Create anti affinity rule for the Kubernetes master VMs #
resource "vsphere_compute_cluster_vm_anti_affinity_rule" "master_anti_affinity_rule" {
  count               = "${var.vsphere_enable_anti_affinity == "true" ? 1 : 0}"
  name                = "${var.vm_name_prefix}-master-anti-affinity-rule"
  compute_cluster_id  = "${data.vsphere_compute_cluster.cluster.id}"
  virtual_machine_ids = ["${vsphere_virtual_machine.master.*.id}"]

  depends_on = ["vsphere_virtual_machine.master"]
}
```

I'd remove these resources from the script. Also `parent_resource_pool_id` was probably declared obsolete, at least there is nothing in the documentation: https://www.terraform.io/docs/providers/vsphere/d/resource_pool.html. See https://github.com/sguyennet/terraform-vsphere-kubespray/issues/26

**Comment by Harsun** [comment]
Ohh ok. I am not an expert in VCenter, but I assume it can work as you
expected.
The best thing, in this case, you can test it and see results.



*Karakas, Gokhan*

Freelance IT advisor, Security Investigator
*Mobile Turkey: +90 506 904 90 59*

**Comment by cbg-teamomni** [comment]
I have a single ESXi server under a vCenter and would like to know if I can deploy without a cluster.

**Comment by semaph0r** [comment]
I'll test in the next few days, if it is possible to refactor the scripts for deployment of the k8s-cluster without a dependency of a compute cluster in vCenter

**Comment by cbg-teamomni** [comment]
Thanks in advance. It would be handy especially for prototyping and demo purposes.
